Added "extension" option

surjikal opened the pull request:
Added an option used to specify the extension of the output file. This is useful when you want to set an extension that is not `.js` or `.html` (`.xml`, for example), or when you want to remove the extension part completely (e.g. when compiling handlebars templates written in jade, s.t. `file.handlebars.jade` becomes `file.handlebars` after it is compiled).

If both the `extension` and `client` options are set, then the specified extension is used.
If the `extension` option is not set, then the extension is determined by the `client` option.

surjikal committed:
Added option to set target's extension

surjikal commented:
Hm, I have a bug in the empty string case. I'll fix it.

surjikal commented:
There, fixed and squashed.

phated commented:
Can you add it to examples?  I will also be adding tests based on another pull request's test (soon) so if you could mimic that test, that would be awesome (but not required).

surjikal committed:
Added example for extension option

surjikal committed:
Added 'locals' option, cleaned up the 'compile' helper a bit.

Before, the 'options' object was passed as the locals. Now, there's
a dedicated 'locals' object. This change breaks backwards compatibility.

The 'locals' option can either be a function or an object.

surjikal committed:
Merge branch 'locals'

Conflicts:
	README.md
	tasks/jade.js

surjikal commented:
Hm, I pushed this to my master branch instead of a feature branch.. I'll try to fix this.

surjikal commented:
Don't think I can modify.. I'll close this pull request and create a new one with the right branch.
